bundler/lib/bundler/plugin.rb

```diff
@@ -62,7 +62,10 @@ def uninstall(names, options)
       if names.any?
         names.each do |name|
           if index.installed?(name)
-            Bundler.rm_rf(index.plugin_path(name))
+            path = index.plugin_path(name).to_s
+            if path.start_with?("#{Plugin.root}/")
```

Comment:
I did my best to find better way to find out the origin of the plugin. It is most likely not possible without extending plugin index functionality. Would it be at least possible to move this logic into `Index` method with proper comment? Btw. this still removes plugin if using `path` inside `Plugin.root` (even I have no idea why anyone would do that).

```ruby
# for example (btw. I'm sure there is better name than local_plugin?)
Bundler.rm_rf(index.plugin_path(name)) unless index.local_plugin?(name)
```

Reply:
I named it `installed_in_plugin_root?`